### PR TITLE
samples: Align partitioning to free space

### PR DESCRIPTION
**Pull request by hellesvik-nordic**
Remove two EMPTY_ partitions by aligning MCUboot inside SPU regions.

### CODE REVIEW

**Review by mswarowsky — APPROVED** [review]
good catch, I started the CI as well to be sure we don't break anything

**Review by SebastianBoe — APPROVED** [review]

File: samples/tfm/tfm_psa_template/child_image/mcuboot/prj.conf [hunk 1]
```diff
@@ -24,3 +24,6 @@ CONFIG_BOOT_SIGNATURE_TYPE_ECDSA_P256=y
 CONFIG_BOOT_SIGNATURE_TYPE_RSA=n
 ### Use minimal C library instead of the Picolib
 CONFIG_MINIMAL_LIBC=y
+
+### Make MCUboot 0x200 smaller to make it fit inside an SPU region
+CONFIG_PM_PARTITION_SIZE_MCUBOOT=0xbe00
```
[the remark this answers is not in the document]
Comment (hellesvik-nordic) [reply samples/tfm/tfm_psa_template/child_image/mcuboot/prj.conf:29]: > Why is this not being done in a static PM file? 

I work in tech support, so from my perspective (and how we see users do things), static configuration very quickly becomes hard to keep control over and get correct. With Kconfigs the build system will take care of stuff making it easier for users.

Does Sysbuild have any Kconfig or other non pm_static.yml way to define partitions or does it only support PM static?

[the remark this answers is not in the document]
Comment (nordicjm) [reply samples/tfm/tfm_psa_template/child_image/mcuboot/prj.conf:29]: Just got the branch and checked out this change with sysbuild and to my surprise it does work, must be something @tejlmand implemented that I was not aware of, will approve and merge

**Review by nordicjm — APPROVED** [review]

File: samples/tfm/tfm_psa_template/child_image/mcuboot/prj.conf (hunk 1 above)
[the remark this answers is not in the document]
Comment (hellesvik-nordic) [reply samples/tfm/tfm_psa_template/child_image/mcuboot/prj.conf:29]: > to my surprise it does work

So I can understand stuff: What do you mean work here exactly? 
My change is one of optimization. The code should work both with and without it. My guess is that it builds both with and without Sysbuild but that partitioning will be different.

[the remark this answers is not in the document]
Comment (nordicjm) [reply samples/tfm/tfm_psa_template/child_image/mcuboot/prj.conf:29]: With this change applied to `sysbuild/mcuboot/prj.conf` the MCUboot partition is 0x200 bytes smaller. I thought all Kconfig sizes were dropped in the sysbuild port but seems that is not the case

**Pull request merged**